test/analysis/headerCheck/src/main.cpp

```diff
@@ -20,7 +20,16 @@
  */
 
 #define BOOST_TEST_MODULE headerCheck
+
+#include <boost/predef.h>   // BOOST_COMP_CLANG
+#if BOOST_COMP_CLANG
+    #pragma clang diagnostic push
+    #pragma clang diagnostic ignored "-Wunused-parameter"
+#endif
 #include <boost/test/unit_test.hpp>
+#if BOOST_COMP_CLANG
+    #pragma clang diagnostic pop
```

Comment:
Most of the time the warning about unknown pragmas is not useful. However, in the case of alpaka that defines many macros containing pragmas, the warning most probably shows a bug within alpaka. Therefore I would not want to fully disable it.